libs/langserve/tests/unit_tests/test_encoders.py

```diff
@@ -0,0 +1,122 @@
+import json
+from typing import Any
+
+import pytest
+from langchain.schema.messages import (
+    HumanMessage,
+    SystemMessage,
+)
+
+from langserve.serialization import simple_dumps, simple_loads
+
+
+@pytest.mark.parametrize(
+    "data, expected_json",
+    [
+        # Test with python primitives
+        (1, 1),
+        ([], []),
+        ({}, {}),
+        ({"a": 1}, {"a": 1}),
+        (
+            {"output": [HumanMessage(content="hello")]},
+            {
+                "output": [
+                    {
+                        "content": "hello",
+                        "additional_kwargs": {},
+                        "type": "human",
+                        "example": False,
+                    }
+                ]
+            },
+        ),
+        # Test with a single message (HumanMessage)
+        (
+            HumanMessage(content="Hello"),
+            {
+                "additional_kwargs": {},
+                "content": "Hello",
+                "example": False,
+                "type": "human",
+            },
+        ),
+        # Test with a list containing mixed elements
+        (
+            [HumanMessage(content="Hello"), SystemMessage(content="Hi"), 42, "world"],
+            [
+                {
+                    "additional_kwargs": {},
+                    "content": "Hello",
+                    "example": False,
+                    "type": "human",
+                },
+                {"additional_kwargs": {}, "content": "Hi", "type": "system"},
+                42,
+                "world",
+            ],
+        ),
+        # # Attention: This test is not correct right now
```

Comment:
Why doesnt this work?

Reply:
De-serialization shouldn't do th eorrect thing because there's no way to distinguish a chunk from a complete message

With that said the test might actually work -- depending on how pydantic tests for equality (attributes or class names) -- I don't think I test just added the test and commented it out :smile: